test/box-luatest/gh_8817_bug_set_language_shared_test.lua

```diff
@@ -0,0 +1,80 @@
+local t = require('luatest')
+local fio = require('fio')
+local popen = require('popen')
+local server = require('luatest.server')
+
+local g = t.group()
+
+-- Create server test instance.
+g.before_all(function()
+    g.server = server:new({alias = 'test-gh-8817-server'})
+    g.server:start()
+    -- Get port for clients.
+    g.server_port = g.server:exec(function()
+        return require('console').listen(0):name()['port']
+    end)
+end)
+
+-- Stop server test instance.
+g.after_all(function()
+    g.server:stop()
+end)
+
+-- Checks that the language setting is not shared between clients.
+g.test_set_language_not_shared_between_clients = function()
+    -- Find tarantoolctl.
+    -- Probably test-run knows the path to tarantoolctl.
+    local TARANTOOLCTL_PATH = os.getenv('TARANTOOLCTL')
+    if TARANTOOLCTL_PATH == nil then
+        -- If test-run dose not know the path to tarantoolctl.
+        -- Assume that current directory is tarantool's build directory.
+        local BUILDDIR = os.getenv('BUILDDIR')
+        TARANTOOLCTL_PATH = ('%s/extra/dist/tarantoolctl'):format(BUILDDIR)
+        if not fio.path.exists(TARANTOOLCTL_PATH) then
+            error("Can't find tarantoolctl")
+        end
+    end
+
+    local cmd_tarantoolctl_connect = {
+        TARANTOOLCTL_PATH,
+        'connect',
+        tostring(g.server_port),
+    }
+
+    -- First connection will be established in a separate process.
+    local ph_connect_first_client = popen.new(cmd_tarantoolctl_connect, {
+        stdin  = popen.opts.PIPE,
+        stdout = popen.opts.PIPE,
+        stderr = 'devnull',
+    })
+
+    ph_connect_first_client:write("\\set language sql")
+    ph_connect_first_client:shutdown({stdin = true})
+
+    local result = ph_connect_first_client:read()
+    ph_connect_first_client:close()
+
+    t.assert_equals(
+        result,
+        "---\n- true\n...\n\n"
+    )
+
+    -- Second connection will be established in a separate process.
+    local ph_connect_second_client = popen.new(cmd_tarantoolctl_connect, {
+        stdin  = popen.opts.PIPE,
+        stdout = popen.opts.PIPE,
+        stderr = 'devnull',
+    })
+
+    ph_connect_second_client:write("\\set language")
+    ph_connect_second_client:shutdown({stdin = true})
+
+    result = ph_connect_second_client:read()
+    ph_connect_second_client:close()
+
+    -- The language changes on the first client did not spread to the second.
+    t.assert_equals(
+        result,
+        "---\n- language: lua\n...\n\n"
+    )
+end
```

Comment:
It is not clear what test should check from the first sight. I suggest splitting test body for test setup (setting up connections to tarantool instance), test teardown (closing connections to tarantool instance) and test logic itself (check language in both connections, change language and check on language again). It will make test more clear and concise.

Reply:
Done, part of the logic has been moved to `g.before_all`

Reply:
Thank you!